app/api/regions.py

```diff
@@ -1,29 +1,100 @@
 from fastapi import APIRouter, Depends, HTTPException, status
 from sqlalchemy.orm import Session
 from typing import List
+import requests
+import asyncpg
+import logging
 
 from app.db.database import get_db
 from app.api.auth import get_current_user_from_auth
-from app.schemas.models import Region, RegionCreate, RegionResponse, User, RegionUpdate
-from app.db.models import DBRegion, DBPrivateAIKey
+from app.schemas.models import Region, RegionCreate, RegionResponse, User, RegionUpdate, TeamSummary
+from app.db.models import DBRegion, DBPrivateAIKey, DBTeamRegion, DBTeam
+from app.core.security import check_system_admin
+
+logger = logging.getLogger(__name__)
 
 router = APIRouter(
     tags=["regions"]
 )
 
-@router.post("", response_model=Region)
-@router.post("/", response_model=Region)
+async def validate_litellm_endpoint(api_url: str, api_key: str) -> bool:
+    """
+    Validate LiteLLM endpoint by making a test request to the health endpoint.
+
+    Args:
+        api_url: The LiteLLM API URL
+        api_key: The LiteLLM API key
+
+    Returns:
+        bool: True if validation succeeds, raises HTTPException if it fails
+    """
+    try:
+        # Test the LiteLLM health endpoint
+        response = requests.get(
+            f"{api_url}/health/liveliness",
+            headers={"Authorization": f"Bearer {api_key}"},
+            timeout=10
+        )
+        response.raise_for_status()
+        logger.info(f"LiteLLM endpoint validation successful for {api_url}")
+        return True
+    except requests.exceptions.RequestException as e:
+        error_msg = str(e)
+        if hasattr(e, 'response') and e.response is not None:
+            try:
+                error_details = e.response.json()
+                error_msg = f"Status {e.response.status_code}: {error_details}"
+            except ValueError:
+                error_msg = f"Status {e.response.status_code}: {e.response.text}"
+        logger.error(f"LiteLLM endpoint validation failed for {api_url}: {error_msg}")
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"LiteLLM endpoint validation failed: {error_msg}"
+        )
+
+async def validate_database_connection(host: str, port: int, user: str, password: str) -> bool:
+    """
+    Validate database connection by attempting to connect to PostgreSQL.
+
+    Args:
+        host: Database host
+        port: Database port
+        user: Database admin user
+        password: Database admin password
+
+    Returns:
+        bool: True if validation succeeds, raises HTTPException if it fails
+    """
+    try:
+        # Attempt to connect to the database
+        conn = await asyncpg.connect(
+            host=host,
+            port=port,
+            user=user,
+            password=password
+        )
```

Comment:
assuming this doesn't need to check any permissions? e.g. by making a select statement or similar.

edit: I see it does attempt a query shortly after this is called, so that's probably fine.